mitodl/mitxpro
# chore!: Upgrade wagtail to 5.x

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes https://github.com/mitodl/hq/issues/2865

#### What's this PR do?
Upgrades Wagtail version to 5.2.1(Latest)

- Disables Pylint as we are facing a string-based model references issue and I tried to resolve it but could not do that easily. We should do that in a separate issue.
- Updates Wagtail imports
- Updates wagtail-factories and wagtail-metadata to fix tests and bugs.
- Creates new migration required for the model changes after the Wagtail upgrade
- Moves all the data migrations to a single migration created at cms/0068.
- Fixes some migration dependencies to resolve migration issues caused by the renaming of PageRevision to Revision.
- Creates a migration at cms/0067 to populate the revision__content_type properly.
- Updates panels in cms/models.py as suggested by the Wagtail upgrade docs.
- Fixes broken tests.
- Removes model-admin for CourseTopic and creates snippet as suggested by Wagtail. Modeladmin is removed from wagtail but we can still use it in case we need it I think it is better to use snippets.
- Fixes courses app migrations to run before the PageRevision model rename migration.

#### How should this be manually tested?

- Checkout master branch
- Create some users, courses, programs, and certificates. Also, create sub-pages for the CoursePage, HomePage, and ProgramPage.
- Integrate xPro with edX (edX integration testing is optional. I have tested locally just to be sure)
- Create enrollments and coupons.
- Create Webinars
- Make sure that you have enough data to test all of the things.
- Now, run `docker-compose stop`
- Check out this branch
- Run `docker-compose build --no-cache`
- Run `docker-compose run --rm web ./manage.py makemigrations`
- Run `docker-compose run --rm web ./manage.py migrate`
- Run `docker-compose run --rm web ./manage.py configure_wagtail`
- Run `docker-compose run --rm web ./manage.py seed_data`
- Run `docker-compose up` and test all of the things that you did on master. Make sure that everything is working and there are no errors.
- Now run `docker-compose down` to remove the old db and run migrations and test all of the things again for a new instance.
- Make sure that the management commands are working fine (I have tested most of the related management commands and they look good to me)


#### Where should the reviewer start?
Review changes, set up your local xPro on the master branch, and then switch back to this branch.

#### Any background context you want to provide?
Pylint is disabled as it was having issues with the String-based model references. We will look into Pylint in a separate issue.

#### What GIF best describes this PR or how it makes you feel?
🥴

#### After Deployment
- [ ] Need to run `./manage.py rebuild_references_index` after the upgrade as suggested in the [docs](https://docs.wagtail.org/en/stable/releases/4.1.html#rebuild-references-index-management-command)
